### PR TITLE
Implement trigger updated option2

### PR DESCRIPTION
**Pull request by Mnakagawa**
Issue #1452

### CODE REVIEW

File: server/hap2/hatohol/hap2_nagios_ndoutils.py [hunk 1]
```diff
@@ -117,6 +118,17 @@ def __parse_url(self, url):
 
         return server, port, database
 
+    def __convert_to_nagios_time(self, hatohol_time):
+        over_second= hatohol_time.split(".")[0]
```
Comment (kz0817) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: missing a space!
over_second= ->
over_second =

File: server/hap2/hatohol/hap2_nagios_ndoutils.py [hunk 2]
```diff
@@ -117,6 +118,17 @@ def __parse_url(self, url):
...
+        time_array = [char for char in over_second]
+        time_array.insert(4, "-")
+        time_array.insert(7, "-")
+        time_array.insert(10, " ")
+        time_array.insert(13, ":")
+        time_array.insert(16, ":")
+
```
Comment (kz0817) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: I seems too lower operations. That degrades the readability and maintainability.
How about making a function that convert a hapi_time to a datetime object in haplib and using strftime().

File: server/hap2/hatohol/hap2_nagios_ndoutils.py [hunk 3]
```diff
@@ -182,12 +194,16 @@ def collect_triggers_and_put(self, fetch_id=None, host_ids=None):
...
             sql += " WHERE %s.host_object_id in ('%s')" % (t2, in_cond)
 
-        # NOTE: The update time in the output is renewed every status
-        #       check in Nagios even if the value is not changed.
-        # TODO by 15.09:
-        #   We should has the previous result and compare it here in order to
-        #   improve performance. Or other columns such as last_state_change in
-        #   nagios_servicestatus might be used.
+        update_type = "ALL"
+        if fetch_id is None:
```
Comment (kz0817) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: The purpose of this condition should be indicated in order to improve readability.

For exampe.

def all_triggers_should_send():
    return fetch_id is None

if all_triggers_should_send():
   ....

File: server/hap2/hatohol/hap2_nagios_ndoutils.py (hunk 2 above)
Comment (Mnakagawa) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: >  it looks this function is only used as making a last_info. 

That's wrong.
It used to create SQL sentence. 
datetime object can not use sql.
So, we should use str() as last_info.

File: server/hap2/hatohol/hap2_nagios_ndoutils.py (hunk 1 above)
Comment (Mnakagawa) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: sure

File: server/hap2/hatohol/hap2_nagios_ndoutils.py (hunk 3 above)
Comment (Mnakagawa) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: k

File: server/hap2/hatohol/hap2_nagios_ndoutils.py (hunk 2 above)
Comment (Mnakagawa) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: The following

https://github.com/project-hatohol/hatohol/blob/implement-trigger-updated-option2/server/hap2/hatohol/hap2_nagios_ndoutils.py#L204

Comment (Mnakagawa) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: I use Timestamp type of HAPI2.0 as a trigger last_info.

Comment (Mnakagawa) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: Issue #1452 deadline is today.
I want to any response.

Comment (kz0817) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: Making an array of character is uncool and may degrade performance.

Comment (Mnakagawa) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: I think so too.
But, python does not have a way to insert a charactor to string.
If you know good idea, please tell me.

Comment (kz0817) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: One thing is to use a datetime object converted from hatohol_time.

Other is to concatenate partial strings of hatohol time such as
x = hatohol_time[0:4] + "-" + hatohol_time[4:6] + "-".
(However, raw numbers are ugly)

File: server/hap2/hatohol/hap2_nagios_ndoutils.py [hunk 4]
```diff
@@ -182,12 +194,19 @@ def collect_triggers_and_put(self, fetch_id=None, host_ids=None):
             in_cond = "','".join(host_ids)
             sql += " WHERE %s.host_object_id in ('%s')" % (t2, in_cond)
 
-        # NOTE: The update time in the output is renewed every status
-        #       check in Nagios even if the value is not changed.
-        # TODO by 15.09:
-        #   We should has the previous result and compare it here in order to
-        #   improve performance. Or other columns such as last_state_change in
-        #   nagios_servicestatus might be used.
+        def all_triggers_should_send():
```
Comment (kz0817) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: This tiny function may be simpler when we use lambda.
<pre>
all_triggers_should_send = lambda: fetch_id is None
</pre>

File: server/hap2/hatohol/hap2_nagios_ndoutils.py (hunk 2 above)
Comment (Mnakagawa) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: > One thing is to use a datetime object converted from hatohol_time.

You mean the following?
hatohol_time(string) -> datatime object (using strptime) -> nagios_time(string. using strftime)

Comment (kz0817) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: YES.

Comment (Mnakagawa) [inline server/hap2/hatohol/hap2_nagios_ndoutils.py]: I understand.
It good idea!

I close this PR and will resend new PR.

**Pull request closed without merging**